### ISSUE TITLE
Layout/ExtraSpacing with ForceEqualSignAlignment aligns on string containing '='

### ISSUE THREAD

**Issue by dmke** [rubocop/rubocop#13509]
Given the following setup:

```yaml
# .rubocop.yml
Layout/ExtraSpacing:
  ForceEqualSignAlignment: true
```

```ruby
# extra_space.rb

# ✅ good
a, b              = val.split('=')
opt.ssh_config[a] = b

# ✅ bad
c, d = val.split(':')
opt.ssh_config[c] = d

# ❌ good, but should be bad
e, f = val.split('=')
opt.ssh_config[e] = f
```

## Expected behavior

Running RuboCop on the `extra_space.rb` should yield two offenses:

- No offense in case (1), as both lines are aligned.
- Case (2) should be an offense as the assignments are not aligned.
- Case (3) should also be an offense, as it is identical to case (2), but with different string contents.

## Actual behavior

Case (3) is incorrectly identified as good:

```console
$ bundle exec rubocop --debug --only Layout/ExtraSpacing extra_space.rb
For /tmp/rubocop: configuration from /tmp/rubocop/.rubocop.yml
Default configuration from /tmp/rubocop/.gems/gems/rubocop-1.69.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /tmp/rubocop/extra_space.rb
C

Offenses:

extra_space.rb:9:19: C: [Correctable] Layout/ExtraSpacing: = is not aligned with the preceding assignment.
opt.ssh_config[c] = d
                  ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.056411404999380466 seconds
```

## Steps to reproduce the problem

```console
$ cd $(mktemp -d)
$ $EDITOR extra_space.rb      # paste contents from above
$ $EDITOR .rubocop.yml        # paste contents from above
$ echo >Gemfile "source 'https://rubygems.org'\n\ngem 'rubocop', require: false"
$ echo >.ruby-version 3.3     # optional, if using rbenv
$ echo >.ruby-gemset  ./.gems # optional, if using rbenv with gemset plugin
$ gem install bundler && bundle install
$ bundle exec rubocop --debug --only Layout/ExtraSpacing extra_space.rb
```

## RuboCop version

```
$ bundle exec rubocop -V
1.69.0 (using Parser 3.3.6.0, rubocop-ast 1.36.2, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
```
